routes/api/users.js

```diff
@@ -131,4 +151,70 @@ router.patch(
   }
 );
 
+router.get("/verify/:verificationToken", async (req, res, next) => {
+  try {
+    const { verificationToken } = req.params;
+    const user = await User.findOne({ verificationToken });
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    user.verify = true;
+    user.verificationToken = null;
+    await user.save();
+
+    res.status(200).json({ message: "Verification successful" });
+  } catch (error) {
+    next(error);
+  }
+});
+router.post("/verify", async (req, res, next) => {
+  try {
+    const { error, value } = resendVerificationSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+    const { email } = value;
+    if (!email) {
+      return res.status(400).json({ message: "Missing required field email" });
+    }
+
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    if (user.verify) {
+      return res
+        .status(400)
+        .json({ message: "Verification has already been passed" });
+    }
+
+    const verificationLink = `${process.env.CLIENT_URL}/users/verify/${user.verificationToken}`;
+    const mailOptions = {
+      from: process.env.GMAIL_USER,
+      to: email,
+      subject: "Account Verification",
+      text: `Click the following link to verify your account: ${verificationLink}`,
+      html: `<p>Click the following link to verify your account: <a href="${verificationLink}">${verificationLink}</a></p>`,
```

Comment:
Tu jes tjuż lepiej. Mozna to ulepszyć w ten sposób że dodasz sobie jakąś funkcję pomocniczą która będzie obsługiwać wysyłanie maili